mockld/polling_service.go

```diff
@@ -117,8 +122,22 @@ func (p *PollingService) pollingHandler(getDataFn func(*PollingService, *http.Re
 		if etag != "" {
 			w.Header().Add("Etag", etag)
 		}
-		w.WriteHeader(http.StatusOK)
-		_, _ = w.Write(data)
+
+		if p.enableGzipCompression && strings.Contains(r.Header.Get("Accept-Encoding"), "gzip") {
+			w.Header().Add("Content-Encoding", "gzip")
+			w.WriteHeader(http.StatusOK)
+			gzipWriter := gzip.NewWriter(w)
+			_, _ = gzipWriter.Write(data)
```

Comment:
I think the `Write` error(s) should be inspected/logged here. If the write or flush fails it might be confusing why.